### ISSUE TITLE
Task manager Global menu plugin

### ISSUE THREAD

**Issue by jleclanche** [lxqt/lxqt#133]
Moving from lxde/lxqt-panel#21

**Comment by paulolieuthier** [comment]
I think that should happen as a third party plugin.

**Comment by nmabhinandan** [comment]
It'd be great to have this feature.

**Comment by fangyq09** [comment]
I appreciate this feature. I agree with JaseIO, the same reason as emce I use Unity.

**Comment by Gibbz** [comment]
kde via kwin now support a kind of global menu, could it be easy to make this work like ubuntu's locally integrated menus?
That is where the window title bar is, on mouse over shows the application menu. 
This is a killer feature for maximizing workspace.

**Comment by GNOME-IS-LIFE** [comment]
Any progress?

**Comment by Lesik** [comment]
I would love to have this as optional feature.

**Comment by tsimonq2** [comment]
I am assigning this to myself, because I am declaring it a feature goal for Lubuntu 19.04. Anyone is welcome to discuss the implementation or implement it, but I'd like to put in the work to get it mergable in time for 19.04.

**Comment by agaida** [comment]
paulolieuthier commented on 17 Apr 2015
> I think that should happen as a third party plugin.

nothing more to say - fully agree

**Comment by tsimonq2** [comment]
I disagree. I'm willing to contribute the code, and it can be enabled optionally.

MATE, Budgie, KDE Plasma, etc. already have this baked in. I don't see any reason why we should treat it as second-class if the code's sane and self-contained.

**Comment by tsimonq2** [comment]
And to be clear - when I say it's a 19.04 goal, that means I want to look into it after 18.10 has shipped. :wink:

**Comment by tsujan** [comment]
* A "good" app menu needs a WM that supports it on its title-bar. KWin does but only inside KDE. LXQt doesn't have a WM of its own.

 * Global menus inside a panel require (1) a horizontal panel (which is a waste of space with all screens I've had in the last 8 years) and (2) an annoying mouse movements for something that's easily accessible on an ordinary menubar.

 * Some apps (like pcmanfm-qt, Cantata, FeatherPad,...) add a optional menu button on their toolbars and hide their menu-bars.

**Comment by tsimonq2** [comment]
> A "good" app menu needs a WM that supports it on its title-bar. KWin does but only inside KDE. LXQt doesn't have a WM of its own.

From my understanding of it, we can just get this from dbus using the platform plugin.

> Global menus inside a panel require (1) a horizontal panel (which is a waste of space with all screens I've had in the last 8 years) and (2) an annoying mouse movements for something that's easily accessible on an ordinary menubar.

YMMV. I disagree. :slightly_smiling_face:  

> Some apps (like pcmanfm-qt, Cantata, FeatherPad,...) add a optional menu button on their toolbars and hide their menu-bars.

Again, my understanding is that we can use dbus.

**Comment by tsujan** [comment]
> Again, my understanding is that we can use dbus.

Yes, dbus is needed. I didn't say it's impossible.

**Comment by agaida** [comment]
and such things would fit best in a third party panel module - ok, we could take it under our umbrella when it is finished and proven to work without flaws.

**Comment by PCMan** [comment]
@tsimonq2 +1 on this one. AFAIK, dbus menu is enough to implement this and we already used it for status icon plugin.
Please create a plugin in lxqtpanel directly but make it optional. :)

**Comment by GaugeK** [comment]
@tsimonq2 I'd love this. Keep us updated if possible. Currently trying out lxqt and it's pretty good, that's the one thing I feel is missing.

**Comment by agaida** [comment]
on a 42" screen distance does matter - trust me. If not, try such a beast yourself (42" 4k) - it will really change your opinion about.

**Comment by tsujan** [comment]
The more it's discussed, the less probably it'll be implemented -- we've had that a few times.

Personally, I dislike it. A developer who likes it should implement it. Until then, discussions won't make a magic.

**Comment by probonopd** [comment]
I'd be strongly interested in a global menu bar. Architecturally, would it have to be implemented as a panel or could it be its own application?

**Comment by Serentty** [comment]
@tsimonq2 Has anything happened with this?

**Comment by probonopd** [comment]
I have written a global menu bar in Qt for [helloSystem](https://hellosystem.github.io/docs/):

![image](https://user-images.githubusercontent.com/2480569/101270613-e4da9480-377a-11eb-8f94-95101d5c582f.png)

**Comment by SViN24** [comment]
while not exactly global menus, what would the devs think of locally integrated menus?
https://youtu.be/nzFXRz-b3Ns
https://invent.kde.org/plasma/breeze/-/merge_requests/126
it essentially hides the menubar and only shows it when the mouse hovers the titlebar of the window.

**Comment by probonopd** [comment]
> while not exactly global menus, what would the devs think of locally integrated menus?

My 2 cents from a UX perspective: 
* They violate Fitts' Law
* They violate Discoverability

So global menus clearly win.

**Comment by tsujan** [comment]
> what would the devs think of locally integrated menus?

LXQt doesn't have a WM of its own. If you apply that patch to breeze and choose KWin as the WM under LXQt, the locally integrated menus should be available (provided that the patch is self-consistent).

To me. they're worse than global menus but that's beside the point.

**Comment by SViN24** [comment]
> > while not exactly global menus, what would the devs think of locally integrated menus?
> 
> My 2 cents from a [usability perspective](https://medium.com/@probonopd/make-it-simple-linux-desktop-usability-part-1-5fa0fb369b42):
> 
>     * They violate Fitts' Law
> 
>     * They negatively impact Affordance (you don't see the menus unless you know that you have to hover over the title bar)
> 
>     * They negatively impact Discoverability (you don't see the names of the menus
> 
>     * ...and have to guess at which coordinates e.g., 'Help' is, so you can't steer the mouse to 'Help' directly. This means you have to navigate to the title bar first, and then navigate to the 'Help' menu. Two movements instead of one.
> 
> 
> So global menus clearly win. By a wide margin!

talking about them as an option aside from global menus.

> > what would the devs think of locally integrated menus?
> 
> LXQt doesn't have a WM of its own. If you apply that patch to breeze and choose KWin as the WM under LXQt, the locally integrated menus should be available (provided that the patch is self-consistent).
> 
> To me. they're worse than global menus but that's beside the point.

Oh i see, however if i remember correctly lxqt allows you to use other window managers, like kwin. could potentially get it working that way when KDE implements it.

**Comment by tsujan** [comment]
> Oh i see, however if i remember correctly lxqt allows you to use other window managers, like kwin. could potentially get it working that way when KDE implements it.

That's almost right. The only thing is that there's no "*other* window managers"; all WMs are treated equally, as far as they're desktop-agnostic. You choose the WM that you want to use with LXQt on the first login. Afterward, you could change it.

**Comment by ryuukk** [comment]
what is the status on this? i'm looking to move away completly from Gnome and GTK applications, so far this is the only piece that prevents me to ditch the xfce panel

**Comment by tsujan** [comment]
> what is the status on this?

Nothing. Apparently, no LXQt dev has found it interesting. But we accept and review patches.
